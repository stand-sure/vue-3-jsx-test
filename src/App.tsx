import { defineComponent } from "vue";
import HelloWorld from "./components/HelloWorld";
// @ts-ignore
import logo from "./assets/logo.png";

export default defineComponent({
  name: "App",
  components: {
    HelloWorld
  },
  render() {
    return (
      <div id="app">
        <img alt="Vue logo" src={logo} />
        <HelloWorld msg={"Welcome to Your Vue.js & TypeScript App using JSX!"} />
      </div>
    );
  }
});
