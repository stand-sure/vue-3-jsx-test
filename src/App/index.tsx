// @ts-ignore
import logo from "../assets/logo.png";
import { Demo } from "../components/demo";

/**
 * @displayName Home
 */
const App = () => {
  return (
    <div class="container mx-auto">
      <img alt="Vue logo" src={logo} class="mx-auto" />
      <Demo />
    </div>
  );
};

export { App };
