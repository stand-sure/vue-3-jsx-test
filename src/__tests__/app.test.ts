import { VNode, RendererNode, RendererElement } from "vue";
import { App } from "../App";

type Node = JSX.Element & VNode<RendererNode, RendererElement>;

//@ts-ignore
import logo from "../assets/logo.png";

describe("App", () => {
  it("should use logo", () => {
    const node = App() as Node;
    const children = node?.children as Node[];
    const img = children?.find(c => c.type === "img");
    const actual = img?.props?.src;
    expect(actual).toBe(logo);
  });

  it("should use Demo", ()=> {
    const node = App() as Node;
    const children = node?.children as Node[];
    const demo = children?.find(c => c.type === "Demo");
    expect(demo).not.toBeNull();
  })
});
