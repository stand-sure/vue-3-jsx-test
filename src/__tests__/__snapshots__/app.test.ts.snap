// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render correctly 1`] = `
Object {
  "__v_isVNode": true,
  "__v_skip": true,
  "anchor": null,
  "appContext": null,
  "children": Array [
    Object {
      "__v_isVNode": true,
      "__v_skip": true,
      "anchor": null,
      "appContext": null,
      "children": null,
      "component": null,
      "dirs": null,
      "dynamicChildren": null,
      "dynamicProps": null,
      "el": null,
      "key": null,
      "patchFlag": 0,
      "props": Object {
        "alt": "Vue logo",
        "class": "mx-auto",
        "src": "test-file-stub",
      },
      "ref": null,
      "scopeId": null,
      "shapeFlag": 1,
      "staticCount": 0,
      "suspense": null,
      "target": null,
      "targetAnchor": null,
      "transition": null,
      "type": "img",
    },
    Object {
      "__v_isVNode": true,
      "__v_skip": true,
      "anchor": null,
      "appContext": null,
      "children": null,
      "component": null,
      "dirs": null,
      "dynamicChildren": null,
      "dynamicProps": null,
      "el": null,
      "key": null,
      "patchFlag": 0,
      "props": null,
      "ref": null,
      "scopeId": null,
      "shapeFlag": 2,
      "staticCount": 0,
      "suspense": null,
      "target": null,
      "targetAnchor": null,
      "transition": null,
      "type": [Function],
    },
  ],
  "component": null,
  "dirs": null,
  "dynamicChildren": null,
  "dynamicProps": null,
  "el": null,
  "key": null,
  "patchFlag": 0,
  "props": Object {
    "class": "container mx-auto",
  },
  "ref": null,
  "scopeId": null,
  "shapeFlag": 17,
  "staticCount": 0,
  "suspense": null,
  "target": null,
  "targetAnchor": null,
  "transition": null,
  "type": "div",
}
`;
